homeassistant/components/hyperion/light.py

```diff
@@ -423,17 +425,20 @@ async def async_turn_on(self, **kwargs: Any) -> None:
             rgb_color = self._rgb_color
 
         # == Set brightness ==
-        if self._brightness != brightness:
-            if not await self._client.async_send_set_adjustment(
-                **{
-                    const.KEY_ADJUSTMENT: {
-                        const.KEY_BRIGHTNESS: int(
-                            round((float(brightness) * 100) / 255)
-                        )
+        if ATTR_BRIGHTNESS in kwargs:
+            brightness = kwargs[ATTR_BRIGHTNESS]
+            for item in self._client.adjustment:
+                if not await self._client.async_send_set_adjustment(
```

Comment:
Perhaps we should be a little less trusting of the input here. If the adjustment item doesn't have KEY_ID in it, skip it.

Reply:
just made a commit to check it.